File: app-node/src/persistence/device.dao.js
class DeviceDao {

    constructor() {
        this._devices = [];
    }

    devices() {
        return this._devices;
    }

    refreshDevices(devices) {
        this._devices = devices //TODO: remove the mutation
    }

    create(device) {
        this._devices.push(device);
        return device;
    }
}

module.exports = DeviceDao;

File: src/main/kotlin/org/andcoe/adf/devices/DeviceDao.kt
package org.andcoe.adf.devices

class DeviceDao {

    private val devicesDb = mutableMapOf<DeviceId, Device>()

    fun devices(): Map<DeviceId, Device> {
        return devicesDb
    }

    fun create(deviceId: DeviceId): Device {
        val device = Device(deviceId)
        devicesDb[deviceId] = device
        return device
    }

    fun remove(deviceId: DeviceId) {
        devicesDb.remove(deviceId)
    }
}